dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/AppClassLoader.java

```diff
@@ -544,6 +599,19 @@ final ByteResourceInformation findClassBytes(String className, String resourceNa
     @Trivial
     @FFDCIgnore(ClassNotFoundException.class)
     protected final Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        // Fail classes which are forbidden.  For example, by a CVE.
+        if ( forbiddenClasses.contains(name) ) {
+            if ( TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled() ) {
+                Tr.debug(tc, "loadClass " + name + " forbidden");
+            }
+            // Does this need to use the 'suggestion' mechanism?
+            // Since this is not a usual class-not-found case,
+            // I'm having it directly throw a simple CNFE.  We
+            // may want to change that to indicate that the class
+            // is designated as to be ignored.
+            throw new ClassNotFoundException(name);
```

Comment:
As the comment suggests, we could give a more detailed error message.  In most cases the JVM is going to catch the CNFE, forget the message and then throw a NoClassDefFoundError with only the class name, so having a more detailed message gives little value besides Class.forName and direct ClassLoader.loadClass calls by the applications.